src/Bookings.js

```diff
@@ -1,21 +1,155 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import SearchResults from "./SearchResults";
+import CustomerProfile from "./CustomerProfile";
+import Spinner from "react-bootstrap/Spinner";
+import BookingForm from "./BookingForm";
 
 const Bookings = () => {
+  const [bookings, setBookings] = useState("");
+  const [buttonId, setButtonId] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState();
   const search = searchVal => {
     console.info("TO DO!", searchVal);
+    let filteredBookings = bookings.filter(booking => {
+      return (
+        booking.firstName.toUpperCase().includes(searchVal.toUpperCase()) ||
+        booking.surname.toUpperCase().includes(searchVal.toUpperCase())
+      );
+    });
+    setBookings(filteredBookings);
   };
+  useEffect(() => {
+    fetch("https://cyf-react.glitch.me/delayed")
+      .then(resp => {
+        if (!resp.ok) {
+          throw new Error(resp.status);
+        } else {
+          return resp.json();
+        }
+      })
+      .then(data => {
+        setBookings(data);
+        setLoading(false);
+      })
+      .catch(error => {
+        console.log(error);
+        setError(error.message);
+      });
+  }, []);
+  function changeButtonId(e) {
+    setButtonId(e.target.id);
+  }
+  const [columnClicked, setColumnClicked] = useState([
+    false,
+    false,
+    false,
+    false
+  ]);
 
-  return (
-    <div className="App-content">
-      <div className="container">
-        <Search search={search} />
-        {/* <SearchResults results={FakeBookings} /> */}
+  function columnSorter(e) {
+    let id = parseInt(e.target.id);
+    if (id === 0) {
+      if (columnClicked[0] === false) {
+        setColumnClicked([true, false, false, false]);
```

Comment:
This solution works, but the logic is confusing. I don't think it's necessary to store 4 booleans, since it's only possible for one column to be sorted at one time. Personally I would store the `columnClicked` as a number which stores the ID of the clicked field. So `setColumnClicked(id);`